packages/ng-primitives/tooltip/src/tooltip-trigger/tooltip-trigger.ts

```diff
@@ -225,18 +239,38 @@ export class NgpTooltipTrigger<T = null> implements OnDestroy {
    * Create the overlay that will contain the tooltip
    */
   private createOverlay(): void {
-    const tooltip = this.state.tooltip();
-
-    if (!tooltip) {
-      throw new Error('Tooltip must be either a TemplateRef or a ComponentType');
+    // Determine the content and context based on useTextContent setting
+    const shouldUseTextContent = this.state.useTextContent();
+    let content = this.state.tooltip();
+    let context: Signal<T | string | undefined> = this.state.context;
+
+    if (!content) {
+      if (!shouldUseTextContent) {
+        throw new Error(
+          '[ngpTooltipTrigger]: Tooltip must be a string, TemplateRef, or ComponentType. Alternatively, set useTextContent to true if none is provided.',
+        );
+      }
+
+      const textContent = this.trigger.nativeElement.textContent?.trim() || '';
+      if (ngDevMode && !textContent) {
+        console.warn(
+          '[ngpTooltipTrigger]: useTextContent is enabled but trigger element has no text content',
+        );
+        return;
+      }
+      content = NgpTooltipTextContentComponent;
+      context = signal(textContent);
+    } else if (typeof content === 'string') {
```

Comment:
@ashley-hunter I wasn't sure if you wanted to introduce a new validators.ts helper for these kinds of assertions, as I saw them inlined in the repo. LMK

Reply:
Yeah, that might be a good idea, there are a few places this kind of check is made so we could reduce duplication.